src/groovy/util/ConfigHelper.groovy

```diff
@@ -29,4 +30,16 @@ class ConfigHelper {
 
     }
 
+    static findAccessRule(String controllerName, String actionName, Object rules) {
+        ArrayList rule = rules.findAll {
+            (it.controller == controllerName && it.actions.contains(actionName)) ||
+                    (it.controller == controllerName && it.actions.contains("*")) ||
+                    (it.controller == "*" && it.actions.contains("*"))
+        }
+        if (rule?.size() > 1) {
+            throw new Exception("There can't be more than one rule specified for this controller and action!")
```

Comment:
This is a good thought, but this could lead to some bad things and might happen frequently. Perhaps instead of an exception we should talk about the precedence rules. What wins by default when there's a conflict like this? My hunch is that the more specific rule should win.

Reply:
I think the easiest way to do this would be to use cascading logic 
```
Map rule = rules.find {
    if (it.controller == controllerName && it.actions.contains(actionName)) return it
    else if (it.controller == controllerName && it.actions.contains("*")) return it
    else if (it.controller == "*" && it.actions.contains("*")) return it
    else return null
}
```
That way you don't have to go back through the list of rules to figure out which one wins using precedence rules.

Reply:
And one thing to note ... your idea about throwing an exception when there are multiple rules makes more sense if there were more than one rule for the matching criteria. 

For example, if we had two rules for "stockRequest:remove" that had different minimum required roles (one was superuser, the other was admin), then we could potentially throw the exception. However, I think it would be better to choose the higher role in this case to avoid the exception. 

Exceptions would be bad because the user didn't do anything wrong so we shouldn't punish them for a sysadmin error. We should just try to limit the damage by taking the most conservative approach i.e. use superuser instead of admin.

Reply:
yeah, I just thought that it didn't have sense to have more than one rule specified for a controller and an action, but you are completely right that it is not user's fault.
I'm wondering how to determine which rule wins, because does this:
```
Map rule = rules.find {
    if (it.controller == controllerName && it.actions.contains(actionName)) return it
    else if (it.controller == controllerName && it.actions.contains("*")) return it
    else if (it.controller == "*" && it.actions.contains("*")) return it
    else return null
}
```
guarantee you that it would find the rule with Superuser instead of Manager if we have something like this (in this order)?
```
 [controller: 'order', actions: ['remove'], accessRules: [ minimumRequiredRole: RoleType.ROLE_MANAGER ]],
 [controller: 'order', actions: ['remove'], accessRules: [ minimumRequiredRole: RoleType.ROLE_SUPERUSER ]]
```